python/ray/monitor.py

```diff
@@ -70,6 +75,31 @@ def parse_resource_demands(resource_load_by_shape):
     return waiting_bundles, infeasible_bundles
 
 
+# Copy pasted from ray.utils to remove dependency on ray.utils.
+def binary_to_hex(identifier):
+    hex_identifier = binascii.hexlify(identifier)
+    if sys.version_info >= (3, 0):
+        hex_identifier = hex_identifier.decode()
+    return hex_identifier
+
+
+async def _get_gcs_address(redis_address: str, redis_password: str) -> str:
+    aioredis_client = await aioredis.create_redis_pool(
+        address="redis://" + redis_address, password=redis_password)
+    gcs_address = await aioredis_client.get("GcsServerAddress")
+    return gcs_address
+
+
+def get_gcs_node_resources_stub(
+        redis_address: str, redis_password: str
+) -> gcs_service_pb2_grpc.NodeResourceInfoGcsServiceStub:
+    loop = asyncio.get_event_loop()
+    gcs_address = loop.run_until_complete(
+        _get_gcs_address(redis_address, redis_password))
+    gcs_channel = aio.insecure_channel(gcs_address)
+    return gcs_service_pb2_grpc.NodeResourceInfoGcsServiceStub(gcs_channel)
```

Comment:
Why do we need asyncio stuff here, can we do it all synchronously for simplicity?

Reply:
I am not sure how you would do it. It seems like the stub is by default asynchronous...

Reply:
Check out internal_api.py for an example of a synchronous call with the stub. The problem is you're using an AIO channel, you can instead use a normal gRPC channel.